static/app/views/ddm/customMetricsEventData.tsx

```diff
@@ -0,0 +1,206 @@
+import {hasDDMFeature} from 'sentry/utils/metrics/features';
+import {formatMRI, parseMRI} from 'sentry/utils/metrics/mri';
+import useOrganization from 'sentry/utils/useOrganization';
+
+function flattenMetricsSummary(
+  metricsSummary: MetricsSummary
+): {item: MetricsSummaryItem; key: string; mri: MRI}[] {
+  return Object.entries(metricsSummary).flatMap(([mri, items]) =>
+    items.map((item, index) => ({item, mri, key: `${mri}${index}`}))
+  );
+}
+
+export function CustomMetricsEventData({
+  metricsSummary,
+}: {
+  metricsSummary: MetricsSummary;
+}) {
+  const organization = useOrganization();
+  const metricsSummaryEntries = flattenMetricsSummary(metricsSummary);
+
+  if (!hasDDMFeature(organization) || metricsSummaryEntries.length === 0) {
+    return null;
+  }
+
+  return (
+    <EventDataSection type="custom-metrics" title={t('Emitted Metrics')}>
+      {metricsSummaryEntries.map(({mri, item, key}) => {
+        return (
+          <Fragment key={key}>
+            <KeyValueList
+              shouldSort={false}
+              data={[
+                {
+                  key: 'name',
+                  subject: t('Name'),
+                  value: <TextOverflow>{formatMRI(mri)}</TextOverflow>,
+                  actionButton: (
+                    <LinkButton
+                      size="xs"
+                      to={getDdmUrl(organization.slug, {
+                        widgets: [
+                          {
+                            mri,
+                            displayType: MetricDisplayType.LINE,
+                            op: getDefaultMetricOp(mri),
+                          },
+                        ],
+                      })}
+                    >
+                      {t('Open in Metrics')}
+                    </LinkButton>
+                  ),
+                },
+                {
+                  key: 'stats',
+                  subject: t('Stats'),
+                  value: <MetricStats item={item} mri={mri} />,
+                },
+                item.tags && Object.keys(item.tags).length > 0
+                  ? {
+                      key: 'tags',
+                      subject: t('Tags'),
+                      value: (
+                        <Tags tags={item.tags} organization={organization} mri={mri} />
+                      ),
+                    }
+                  : null,
+              ].filter((row): row is Exclude<typeof row, null> => Boolean(row))}
+            />
+          </Fragment>
+        );
+      })}
+    </EventDataSection>
+  );
+}
+
+function MetricStats({mri, item}: {item: MetricsSummaryItem; mri: MRI}) {
+  const parsedMRI = parseMRI(mri);
+  const unit = parsedMRI?.unit ?? 'none';
+  const type = parsedMRI?.type ?? 'c';
+
+  const typeLine = t(`Type: %s`, getReadableMetricType(type));
+  // We use formatMetricUsingUnit with unit 'none' to ensure uniform number formatting
+  const countLine = t(`Count: %s`, formatMetricUsingUnit(item.count, 'none'));
```

Comment:
How about we do not show the count if it's just 1? For additional brevity.

Reply:
But then I don't know what the count was 😅

Reply:
I was thinking that if the count is 1, we just show the value. If the count is more, we display how many times it happened and the range, like we already do (min, max, etc.). But maybe I misunderstood things.